### PR TITLE
Add rate limit check support for gitlab

### PR DESCRIPTION
**Pull request by Xuanwo**

### CODE REVIEW

File: tests/test_gitlab_local.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# MIT licensed
+# Copyright (c) 2013-2017 lilydjwg <lilydjwg@gmail.com>, et al.
+
+import os
+import pytest
+import asyncio
+pytestmark = [pytest.mark.asyncio,
+              pytest.mark.skipif("TRAVIS" in os.environ and os.environ["TRAVIS"] == "true",
```
Comment (lilydjwg) [inline tests/test_gitlab_local.py]: Use `os.environ.get('TRAVIS') == "true"`

File: nvchecker/source/gitlab.py [hunk 2]
```diff
@@ -51,3 +58,16 @@
...
+def check_ratelimit(exc, name):
+  res = exc.response
+  n = int(res.headers.get('RateLimit-Remaining'))
+  if n == 0:
+    reset = int(res.headers.get('RateLimit-Reset'))
+    logger.error('rate limited, resetting at %s. '
+                 'Or get an API token to increase the allowance if not yet'
+                 % time.ctime(reset),
+                 name = name,
+                 reset = reset)
```
Comment (lilydjwg) [inline nvchecker/source/gitlab.py]: We can't get the real reset time. Let's put `(unknown)` there instead.

File: tests/test_gitlab_local.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                                 reason="do not test on travis ci")]
```
Comment (lilydjwg) [inline tests/test_gitlab_local.py]: This reason is useless. Say it "rate-limited per IP". Also, you need to unset the environment variable in this test file (and restore it later).

File: tests/test_gitlab_local.py (hunk 1 above)
Comment (Xuanwo) [reply tests/test_gitlab_local.py]: Fixed.

File: nvchecker/source/gitlab.py (hunk 2 above)
Comment (Xuanwo) [reply nvchecker/source/gitlab.py]: Fixed.

File: tests/test_gitlab_local.py (hunk 3 above)
Comment (Xuanwo) [reply tests/test_gitlab_local.py]: > you need to unset the environment variable in this test file (and restore it later).

Unset "TRAVIS" environment variable ?

Comment (lilydjwg) [reply tests/test_gitlab_local.py]: The gitlab token.

Comment (Xuanwo) [reply tests/test_gitlab_local.py]: Got it, fix is coming.

**Comment by Xuanwo** [comment]
@lilydjwg  All changes requested have been done, and the gitlab token detect logic has been rewritten, please take a look.

File: tests/test_gitlab_local.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    with unset_gitlab_token_env():
+        assert await get_version("example", {"gitlab": "gitlab-org/gitlab-test", "use_max_tag": 1}) == "v1.1.0"
+
+async def test_gitlab_max_tag_with_ignored_tags(get_version):
+    with unset_gitlab_token_env():
+        ver = await get_version("example",
+                                {"gitlab": "gitlab-org/gitlab-test", "use_max_tag": 1, "ignored_tags": "v1.1.0"})
+        assert ver == "v1.0.0"
+
+async def test_gitlab_over_limit(get_version):
```
Comment (lilydjwg) [inline tests/test_gitlab_local.py]: We'd better not do this. Also, nothing is asserted.

**Pull request merged**